packages/trie/src/baseTrie.ts

```diff
@@ -39,24 +39,30 @@ export type FoundNodeFunction = (
  * The API for the base and the secure interface are about the same.
  */
 export class Trie {
-  /** The backend DB */
-  db: DB
   /** The root for an empty trie */
   EMPTY_TRIE_ROOT: Buffer
-  private _root: Buffer
   protected lock: Semaphore
 
+  /** The backend DB */
+  db: DB
+  private _root: Buffer
+  private _deleteFromDB: boolean
+
   /**
    * test
    * @param db - A [levelup](https://github.com/Level/levelup) instance. By default (if the db is `null` or
    * left undefined) creates an in-memory [memdown](https://github.com/Level/memdown) instance.
    * @param root - A `Buffer` for the root of a previously stored trie
+   * @param deleteFromDB - Delete nodes from DB on delete operations (disallows switching to an older state root) (default: `false`)
    */
-  constructor(db?: LevelUp | null, root?: Buffer) {
+  constructor(db?: LevelUp | null, root?: Buffer, deleteFromDB: boolean = false) {
```

Comment:
I suggest we introduce a `TrieOptions` interface here like our other monorepo packages to remain consistent, just because I feel we could add more options in the future and would be easier to manage.

Reply:
Had planned this in for v6 and added a note #1024 since we already have these two optional parameters floating around (so `db` and `root`, or would you keep these separate for some reasons?). Or would you add this option nevertheless now and just do the additional inclusions on v6 (might be the more flexible solution)?

Reply:
ah okay nice, I think that's fine we can do it for v6 and keep this as is 👍